### PR TITLE
Add documentation to add GA4 begin_checkout event

### PR DESCRIPTION
**Pull request by stephaniehobson**
## One-line summary

Add documentation to add GA4 begin_checkout event

## Significant changes and points to review

- reorganized docs to give Pocket a separate page from moz-org, changing all heading levels
- noted some outdated documentation in the data-attribute section
- added a section on GA4 with the begin_checkout info
- added references to it from the VPN docs.

## Issue / Bugzilla link

#13012 

## Testing

[Compile the docs and run them locally.](https://bedrock.readthedocs.io/en/latest/install.html#documentation)

http://127.0.0.1:8100/attribution/0001-analytics.html

### CODE REVIEW

File: docs/attribution/0001-analytics.rst [hunk 1]
```diff
@@ -5,15 +5,11 @@
...
+Mozorg analytics
 =================
 
-***********
-Mozorg Mode
-***********
-
 Google Tag Manager (GTM)
-------------------------
+************************
```
Comment (nathan-barrett) [inline docs/attribution/0001-analytics.rst:12]: **Question**: Why the change from the previous symbol (`-`) to `*`

**Review by nathan-barrett — APPROVED** [review]
great work! LGTM r+

File: docs/attribution/0001-analytics.rst (hunk 1 above)
Comment (stephaniehobson) [reply docs/attribution/0001-analytics.rst:12]: To change the heading levels. Everything got bumped up a level when I removed the Pocket stuff.
